packages/core/src/indexing/local-index-api.ts

```diff
@@ -174,19 +173,12 @@ export class LocalIndexApi implements IndexApi {
   }
 
   async init(): Promise<void> {
-    await this.databaseIndexApi?.init()
-    // FIXME: CDB-2132 - Fragile DatabaseApi initialisation
-    await this.populateDatabaseApiInternalState()
-  }
-
-  /*
-  Gets currently indexed models from the database api and calls index's api indexModels()
-  which creates index model args and passes them back to the database api, so that it can populate
-  its internal state
-   */
-  async populateDatabaseApiInternalState(): Promise<void> {
-    const modelsToIndex = this.databaseIndexApi?.getActiveModelsToIndex()
-    await this.indexModels(modelsToIndex)
+    if (this.databaseIndexApi) {
```

Comment:
nitpick: would prefer early return with
```
if(!this.databaseIndexApi) {
  return
}
```

Early returns are nice for avoiding extra indentation layers and improving readability IMO